src/libraries/System.Runtime.Numerics/src/System/Numerics/NumericsHelpers.cs

```diff
@@ -2,32 +2,20 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
-using System.Runtime.InteropServices;
 
 namespace System.Numerics
 {
-    [StructLayout(LayoutKind.Explicit)]
-    internal struct DoubleUlong
-    {
-        [FieldOffset(0)]
-        public double dbl;
-        [FieldOffset(0)]
-        public ulong uu;
-    }
-
     internal static class NumericsHelpers
     {
         private const int kcbitUint = 32;
 
         public static void GetDoubleParts(double dbl, out int sign, out int exp, out ulong man, out bool fFinite)
         {
-            DoubleUlong du;
-            du.uu = 0;
-            du.dbl = dbl;
+            ulong bits = BitConverter.DoubleToUInt64Bits(dbl);
 
-            sign = 1 - ((int)(du.uu >> 62) & 2);
-            man = du.uu & 0x000FFFFFFFFFFFFF;
-            exp = (int)(du.uu >> 52) & 0x7FF;
+            sign = 1 - ((int)(bits >> 62) & 2);
```

Comment:
Is `bits` really 0 here?

Reply:
DoubleUlong is a union, `bdl` is set later so `uu` is the bits of double